### DISCUSSION TITLE
How to remove unused chunk files after unused export pruning?

### DISCUSSION THREAD

**Discussion by gkjohnson** [webpack/webpack#13029] [Q&A]
Consider the following set of files. Notice that the "unusedFunction" in `dependency.js` dynamically imports `dynamic.js` and will be removed in a final production build:

```js
// dynamic.js

export function func() { console.log( 'dynamic import log' ); }
```

```js
// dependency.js
export function usedFunction() { console.log( 'USED' ); }

export function unusedFunction() {

    import( './dynamic.js' ).then( mod => mod.func() );

}
```

```js
// index.js
import { usedFunction } from './dependency.js';

usedFunction();
```

When the final production build is produced two files are created: 1 file with the `index.js` code and used function from `dependency.js` and 1 file for the dynamic import from the unused function (called `815.js` in my case). In the process of pruning unused code from the build, the unused function which imported `815.js` was removed from the bundle and therefore `815.js` is no longer imported anywhere and no longer needed.

Is there any way to get webpack to remove these now-orphaned chunks after unused export pruning has occurred? If not is there a way to detect them so I can remove them manually?

Thank you!

**Comment by nyngwang** [comment]
lol, did you solve it? I'm also interested in the solution.

**Reply by vankop** [reply]
you need to set `sideEffects` flag for this.
